AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Dependency/AuthenticationProviderAdapter+DeleteUser.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+#if COCOAPODS
+import AWSMobileClient
+#else
+import AWSMobileClientXCF
+#endif
+
+extension AuthenticationProviderAdapter {
+
+    func deleteUser(request: AuthDeleteUserRequest, completionHandler: @escaping (Result<Void, AuthError>) -> Void) {
+        awsMobileClient.deleteUser(signOut: false) { [weak self] error in
```

Comment:
Why is `signOut` false rather than true? It's not clear what relationship this argument has to the `globalSignOut` call below on L20, so a comment is in order to explain the apparent discrepancy

Reply:
By default, ASWMobileClient calls signOut internally during deleteUser. For Amplify, we instead call Amplify's signOut function, which contains some higher level logic. This has the advantage of keeping all the Amplify signOut/cleanup code in one place. I added a comment to document this.

Reply:
Updated to handle signout in AWSMobileClient in order to better handle hosted UI sessions.